lib/improver/nbhood/square_kernel.py

```diff
@@ -486,44 +495,40 @@ def _pad_and_calculate_neighbourhood(
 
         Returns:
-            neighbourhood_averaged_cubes (Iris.cube.CubeList):
-                CubeList containing the smoothed field after the square
-                neighbourhood method has been applied to either the input cube,
-                or both the input cube and a mask cube.
+            neighbourhood_averaged_cube (Iris.cube):
+                Cube containing the smoothed field after the square
+                neighbourhood method has been applied.
         """
-        neighbourhood_averaged_cubes = iris.cube.CubeList([])
-        masked_data = len(cubes_to_sum) > 1
-        for cube_to_process in cubes_to_sum:
+
+        for i, cube_to_process in enumerate(cubes_to_sum):
             # Pad the iris cube. This way, the edge effects produced
             # by the vectorisation of the 4-point method will appear outside
             # our domain of interest. These unwanted points can be trimmed off
             # later.
-            cube_to_process = self.pad_cube_with_halo(
-                cube_to_process, grid_cells_x, grid_cells_y, masked_data)
-            summed_up_cube, nan_masks = self.cumulate_array(
-                cube_to_process)
-            neighbourhood_averaged_cubes.append(
-                self.mean_over_neighbourhood(
-                    summed_up_cube, grid_cells_x, grid_cells_y, nan_masks))
-        return neighbourhood_averaged_cubes
+            cubes_to_sum[i] = self.pad_cube_with_halo(
+                cube_to_process, grid_cells_x, grid_cells_y,
+                masked_halo=True)
+        summed_up_cubes, nan_mask = self.cumulate_array(cubes_to_sum)
+        neighbourhood_averaged_cube = (
+            self.mean_over_neighbourhood(
+                summed_up_cubes, grid_cells_x, grid_cells_y, nan_mask))
+        return neighbourhood_averaged_cube
 
     def _remove_padding_and_mask(
-            self, neighbourhood_averaged_cubes, pre_neighbourhood_cubes,
-            cube_name, grid_cells_x, grid_cells_y):
+            self, neighbourhood_averaged_cube,
+            original_cube, mask_cube,
+            grid_cells_x, grid_cells_y):
         """
         Remove the halo from the padded array and apply the mask, if required.
 
         Args:
-            neighbourhood_averaged_cubes (Iris.cube.CubeList):
-                CubeList containing the smoothed field after the square
-                neighbourhood method has been applied to either the input cube,
-                or both the input cube and a mask cube.
-            pre_neighbourhood_cubes (Iris.cube.CubeList):
-                CubeList containing the fields prior to applying neighbourhood
-                processing. This is required to be able to know the original
-                mask cube.
-            cube_name (String):
-                Name of the variable that has been neighbourhooded.
+            neighbourhood_averaged_cube (Iris.cube.Cube):
+                Cube containing the smoothed field after the square
```

Comment:
This should probably mention that the input cube has a halo still applied.

Reply:
Added